core/trie/proof.go

```diff
@@ -87,7 +143,7 @@ func transformNode(tri *Trie, parentKey *Key, sNode storageNode) (*Edge, *Binary
 	}
 
 	rightHash := rNode.Value
-	if isEdge(sNode.key, storageNode{node: rNode, key: sNode.node.Right}) {
+	if isEdge(sNode.key, StorageNode{node: rNode, key: sNode.node.Right}) {
```

Comment:
2. Yes, the value is irrelevant here since we just want to know if the left/right child is an edge node, which we can tell using the length of the parent and child keys